content/tutorials/jwt-authentication.textile

```diff
@@ -273,41 +276,57 @@ The last part is to add the logic into this form. You'll instantiate Ably's Real
 ```[html]
 <script type="text/javascript">
 console.log('On login page now')
+
+
 function login(e) {
-    e.preventDefault();
-    console.log('Fetching JWT token from auth server')
-    const realtime = new Ably.Realtime({ authUrl: '/auth' });
-    realtime.connection.once('connected', function () {
-    console.log('Client connected to Ably using JWT')
-    alert("Client successfully connected to Ably using JWT auth")
-    })
-}
+  e.preventDefault();
+
+  const username = document.querySelector("#username").value;
+  const password = document.querySelector("#password").value;
+
+  if (!`${username}`.trim() || !`${password}`.trim()) {
+    return alert("Error! Require username and password");
+  }
+
+  // Send user details to the auth server
+  const authUrl = `/auth?clientId=${username}:${password}`;
+  const realtime = new Ably.Realtime({ authUrl });
+
+  console.log("Fetching JWT token from auth server", username, password);
+
+  realtime.connection.once("connected", () => {
+    const { tokenDetails } = realtime.auth;
+    console.log("Client connected to Ably using JWT", tokenDetails);
+    alert("Client successfully connected Ably using JWT auth");
+  });
 </script>
 ```
 
 The code begins by instantiating Ably's Realtime client library, then you assign the path of auth server, @/auth@ to the @authUrl@ parameter. The @/auth@ path of on auth server is responsible for signing a JWT using the private API key and then returning back the @tokenId@, the JWT itself.
 
+It is worth noticing that we also sending the user input to the @authUrl@ as a querystring @(/auth?clientId=${username}:${password})@. This is so that our authentication server can validate the user requesting the JWT. In this simple example we will not cover that, and this detail is to demonstrate how you might send user input to your server to control user connections.
```

Comment:
We said this in an earlier step. Probably better to lose this sentence and just remind the reader what they did in that step?

Reply:
true, true.